lib/Doctrine/DBAL/Driver/V3CompatStatement.php

```diff
@@ -0,0 +1,203 @@
+<?php
+
+namespace Doctrine\DBAL\Driver;
+
+use Doctrine\DBAL\Abstraction\V3Result;
+use Doctrine\DBAL\Exception;
+use IteratorAggregate;
+use PDO;
+use Traversable;
+
+class V3CompatStatement implements IteratorAggregate, ResultStatement, V3Result
+{
+    /** @var ResultStatement */
+    private $stmt;
+
+    public function __construct(ResultStatement $stmt)
+    {
+        $this->stmt = $stmt;
+    }
+
+    /**
+     * @return ResultStatement
+     */
+    public function getIterator()
+    {
+        return $this->stmt;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function closeCursor()
+    {
+        return $this->stmt->closeCursor();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function columnCount()
+    {
+        return $this->stmt->columnCount();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function setFetchMode($fetchMode, $arg2 = null, $arg3 = null)
+    {
+        return $this->stmt->setFetchMode($fetchMode, $arg2, $arg3);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function fetch($fetchMode = null, $cursorOrientation = PDO::FETCH_ORI_NEXT, $cursorOffset = 0)
+    {
+        return $this->stmt->fetch($fetchMode, $cursorOrientation, $cursorOffset);
```

Comment:
I don't know if it's applicable to this case but in some cases, just passing the arguments down the line doesn't work because the underlying driver relies on the number of arguments passed, not their values: https://github.com/doctrine/dbal/blob/84553e361d7ab15b16c45e1a08572d07e8a992e0/lib/Doctrine/DBAL/Driver/PDOStatement.php#L127-L136

The fact that the build is passing means that most likely this case isn't tested.

Reply:
Not sure to fully understand in which case it could cause an issue. Anyway, all drivers statements of the library have been migrated on forward compatible Result interface, so they will not be wrapped in the compatibility statement object.

https://github.com/doctrine/dbal/blob/c4e4ce67bde3d407db27b157306373d52a6b0bec/lib/Doctrine/DBAL/Connection.php#L1308-L1315